integrations/docker/images/chip-cert-bins/Dockerfile

```diff
@@ -0,0 +1,273 @@
+    && curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py \
+    && python3.9 get-pip.py \
+    && update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1 \
+    && rm -rf /var/lib/apt/lists/ \
+    && : # last line
+
+RUN set -x \
+    && pip3 install attrs coloredlogs PyGithub pygit future portpicker mobly click cxxfilt ghapi pandas tabulate \
+    && : # last line
+
+# build and install gn
+RUN set -x \
+    && git clone https://gn.googlesource.com/gn \
+    && cd gn \
+    && python3 build/gen.py \
+    && ninja -C out \
+    && cp out/gn /usr/local/bin \
+    && cd .. \
+    && rm -rf gn \
+    && : # last line
+
+# Install bloat comparison tools
+RUN set -x \
+    && git clone https://github.com/google/bloaty.git \
+    && mkdir -p bloaty/build \
+    && cd bloaty/build \
+    && cmake ../ \
+    && make -j8 \
+    && make install \
+    && cd ../.. \
+    && rm -rf bloaty \
+    && : # last line
+
+# NodeJS: install a newer version than what apt-get would read
+# This installs the latest LTS version of nodejs
+RUN case ${TARGETPLATFORM} in \
+        "linux/amd64") \
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-x64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-x64.tar.xz \
+			&& mv node-v12.19.0-linux-x64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-x64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        "linux/arm64")\
+            set -x \
+			&& mkdir node_js \
+			&& cd node_js \
+			&& wget https://nodejs.org/dist/v12.19.0/node-v12.19.0-linux-arm64.tar.xz \
+			&& tar xfvJ node-v12.19.0-linux-arm64.tar.xz \
+			&& mv node-v12.19.0-linux-arm64 /opt/ \
+			&& ln -s /opt/node-v12.19.0-linux-arm64 /opt/node \
+			&& ln -s /opt/node/bin/* /usr/bin \
+			&& cd .. \
+			&& rm -rf node_js \
+			;; \
+        *) ;; \
+    esac
+
+# Stage 1.5: Bootstrap Matter.
+# Set python to python3
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
+
+RUN mkdir /root/connectedhomeip
+RUN git clone --depth 1 --single-branch --branch ${BRANCH} https://github.com/project-chip/connectedhomeip.git /root/connectedhomeip
+WORKDIR /root/connectedhomeip/
+RUN scripts/build/gn_bootstrap.sh
+RUN gn gen out/debug --args='chip_mdns="platform" chip_inet_config_enable_ipv4=false'
+RUN ninja -C out/debug
+
+# Stage 2: Build.
+from chip-build-cert as chip-build-cert-bins
+SHELL ["/bin/bash", "-c"]
+# Records Matter SDK commit hash to include in the image.
+RUN git rev-parse HEAD > /root/.sdk-sha-version
+RUN case ${TARGETPLATFORM} in \
+        "linux/amd64") \
+            set -x \
+			&& source scripts/activate.sh \
+			&& scripts/build/build_examples.py --target linux-x64-all-clusters-ipv6only build \
+			&& mv out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app out/chip-all-clusters-app \
+			&& scripts/build/build_examples.py --target linux-x64-bridge-ipv6only build \
```

Comment:
Good suggestion, I'll try this.

Reply:
Welp, that did not work as expected.
`#26 472.6 2022-07-26 20:41:50 INFO    Copying /root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app into chip-all-clusters-app                                                                                                     
#26 472.6 Traceback (most recent call last):                                                                                                                                                                                                                 
#26 472.6   File "/root/connectedhomeip/scripts/build/build_examples.py", line 228, in <module>                                                                                                                                                              
#26 472.6     main()                                                                                                                                                                                                                                         
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__                                                                                                                        
#26 472.6     return self.main(*args, **kwargs)                                                                                                                                                                                                              
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1055, in main                                                                                                                            
#26 472.6     rv = self.invoke(ctx)                                                                                                                                                                                                                          
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1688, in invoke                                                                                                                          
#26 472.6     rv.append(sub_ctx.command.invoke(sub_ctx))                                                                                                                                                                                                     
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke                                                                                                                          
#26 472.6     return ctx.invoke(self.callback, **ctx.params)                                                                                                                                                                                                 
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke                                                                                                                           
#26 472.6     return __callback(*args, **kwargs)                                                                                                                                                                                                             
#26 472.6   File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func                                                                                                                    
#26 472.6     return f(get_current_context(), *args, **kwargs)                                                                                                                                                                                               
#26 472.6   File "/root/connectedhomeip/scripts/build/build_examples.py", line 221, in cmd_build                                                                                                                                                             
#26 472.6     context.obj.CopyArtifactsTo(copy_artifacts_to)                                                                                                                                                                                                 
#26 472.6   File "/root/connectedhomeip/scripts/build/build/__init__.py", line 92, in CopyArtifactsTo                                                                                                                                                        
#26 472.6     builder.CopyArtifacts(os.path.join(path, builder.identifier))                                                   
#26 472.6   File "/root/connectedhomeip/scripts/build/builders/builder.py", line 116, in CopyArtifacts                        
#26 472.6     shutil.copyfile(source_name, target_full_name)                                                                  
#26 472.6   File "/usr/lib/python3.10/shutil.py", line 234, in copyfile                                                       
#26 472.6     raise SameFileError("{!r} and {!r} are the same file".format(src, dst))                                         
#26 472.6 shutil.SameFileError: '/root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app' and '/root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app' are the same file`

Reply:
For readability, lines below.

I think you cannot omit the `--copy-artifacts-to out` parameter and it defaults to put them in `out`


```
#26 472.6 2022-07-26 20:41:50 INFO Copying /root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app into chip-all-clusters-app 
#26 472.6 Traceback (most recent call last): 
#26 472.6 File "/root/connectedhomeip/scripts/build/build_examples.py", line 228, in <module> 
#26 472.6 main() 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__ 
#26 472.6 return self.main(*args, **kwargs) 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1055, in main 
#26 472.6 rv = self.invoke(ctx) 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1688, in invoke 
#26 472.6 rv.append(sub_ctx.command.invoke(sub_ctx)) 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke 
#26 472.6 return ctx.invoke(self.callback, **ctx.params) 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke 
#26 472.6 return __callback(*args, **kwargs) 
#26 472.6 File "/root/connectedhomeip/.environment/pigweed-venv/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func 
#26 472.6 return f(get_current_context(), *args, **kwargs) 
#26 472.6 File "/root/connectedhomeip/scripts/build/build_examples.py", line 221, in cmd_build 
#26 472.6 context.obj.CopyArtifactsTo(copy_artifacts_to) 
#26 472.6 File "/root/connectedhomeip/scripts/build/build/__init__.py", line 92, in CopyArtifactsTo 
#26 472.6 builder.CopyArtifacts(os.path.join(path, builder.identifier)) 
#26 472.6 File "/root/connectedhomeip/scripts/build/builders/builder.py", line 116, in CopyArtifacts 
#26 472.6 shutil.copyfile(source_name, target_full_name) 
#26 472.6 File "/usr/lib/python3.10/shutil.py", line 234, in copyfile 
#26 472.6 raise SameFileError("{!r} and {!r} are the same file".format(src, dst)) 
#26 472.6 shutil.SameFileError: '/root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app' and '/root/connectedhomeip/out/linux-x64-all-clusters-ipv6only/chip-all-clusters-app' are the same file

Reply:
Or maybe use something else than `out` eg. `--copy-artifacts-to cert-bins`

Reply:
Yeah, I know. Boris suggested the --copy-artifacts-to out maybe to help avoid the mv's that I did before. However, the whole reason why I have to do the mv's is because the amd64 and arm64 builds have different paths, but for Stage 3 I need them to be the same.